server/src/main/java/org/elasticsearch/synonyms/SynonymsManagementAPIService.java

```diff
@@ -179,4 +219,26 @@ static Settings settings() {
             .build();
     }
 
+    public enum UpdateSynonymsResult {
```

Comment:
I did some refactoring for PUT. It didn't feel right that the `SynonymsManagementAPIService` depended on the request, so I defined the response enum in the `SynonymsManagementAPIService` and the response now uses it.

Reply:
I like the previous arrangement more, I think `UpdateSynonymsResult` should be a part of `PutSynonymsAction.Response` class and should not be a part of `SynonymsManagementAPIService` class. It is weird to break response across multiple classes.

Reply:
😆 Serves me right for refactoring! 😛 

I feel that having a mutual dependency is wrong. For me, it's good to have dependencies in a single direction, from the external layers (request) to the internal (the service in this case). It feels a bit wrong to have the callee to have a dependency on the caller.

Reply:
😆 Serves me right for refactoring! 😛 

I feel that having a mutual dependency is wrong. For me, it's good to have dependencies in a single direction, from the external layers (request) to the internal (the service in this case). It feels a bit wrong to have the callee to have a dependency on the caller.

Reply:
Thanks for  an extra explanation, I understood your point, and indeed the way you refactored it would be better if `SynonymsManagementAPIService` would be some general class used by many. But I see it as an extension of `Transport{GET/PUT}SynonymsAction`. That's why I don't see the need of these extra classes that makes the code more complex. 

But I don't know, if you strongly feel for having them, we can keep them. I will leave this to you.

Reply:
Discussed offline with @carlosdelest, and we are ok to proceed with the currently implemented approach.